### PR TITLE
Refactor modular

### LINKED ISSUE #71: CLEANUP: Refactor code for modularity
As title says, yo

### PR DESCRIPTION
**Pull request by tiffeli**
- we were doing navigator.geolocation functions a lot so I thought we could definitely abstract that function and be more DRY.

- all the navigator functions are located in utils.js and imported as geoAction in the files.

1. getCurrent = navigator.geolocation.getCurrentPosition

        - takes in a callback that is passed in successfully found coords (object w latitude and longitude)

2. setWatch= navigator.geolocation.watchPosition
        - takes in a callback that is passed in successfully found coords (object w latitude and longitude)

3. clearWatch = navigator.geolocation.clearWatch

        - takes in watchID that you want to clear.
 

**note: remember to save this binding by declaring self = this outside of the function and refer all 'this' as self**. See the refactored functions for example.

- call them by invoking eg:  geoAction.getCurrent()

resolves #71

### CODE REVIEW

File: app/components/Map.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ import baseImg from '../assets/redPin.png';
 import targetImg from '../assets/blackPin.png';
 import { PinCallout } from './PinCallout';
 import PinEditButton from './PinEditButton';
-import { myCurrLoc, currLoc } from '../lib/db/db';
+import * as geoAction from '../lib/orientation/utils';
```
Comment (bignell) [inline app/components/Map.js]: @tiffeli We lost the `myCurrLoc` and `currLoc` variables here, but they're still in use on line 64 (throws an error).

Comment (bignell) [inline app/components/Map.js]: @tiffeli Hmm, it looks you're using `setState` to instantiate at least `currLoc` later, so maybe the fix is as simple as changing line 64 to `this.state.currLoc`.

**Comment by bignell** [comment]
@tiffeli M · O · D · U · L · A · R · I · T · Y ! !

Awesome stuff. Just one bug encountered, see comments.

**Comment by asu91** [comment]
@MERGE

**Pull request merged**